packages/@aws-cdk/aws-lambda/test/test.code.ts

```diff
@@ -81,6 +81,29 @@ export = {
         }
       }, ResourcePart.CompleteDefinition));
       test.done();
+    },
+
+    'codeHash is supported for assets and inline'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const asset = new lambda.Function(stack, 'AssetCode', {
+        code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+        runtime: lambda.Runtime.NODEJS_12_X,
+        handler: 'index.handler'
+      });
+
+      const inline = new lambda.Function(stack, 'InlineCode', {
+        code: lambda.Code.fromInline('boom boom'),
+        runtime: lambda.Runtime.NODEJS_12_X,
+        handler: 'index.handler'
+      });
+
+      // THEN
+      test.deepEqual(asset.codeHash, '9678c34eca93259d11f2d714177347afd66c50116e1e08996eff893d3ca81232');
+      test.deepEqual(inline.codeHash, '58ad733fe276cb7b9b171b0beafad40f43b6d1cd68a1b93327d3655f739dfa0e');
```

Comment:
Also assert that it is `undefined` for s3 asset.

Reply:
done